prow/plugins/dog/dog_test.go

```diff
@@ -96,7 +96,7 @@ func TestFormat(t *testing.T) {
 // Medium integration test (depends on ability to open a TCP port)
 func TestHttpResponse(t *testing.T) {
 	// setup a stock valid request
-	url := "http://localhost/dog.jpg"
+	url := "https://random.dog/2e0de7b5-cc16-4112-8c5e-2c18708cf800.jpg"
```

Comment:
I'm not sure if this is allowable, it is not hermetic. @BenTheElder

Reply:
this will break if the site is down or similar, we can fake out the request locally which will be less flaky..

Reply:
Ok, I will use httptest instead. That was actually my question in the comment...
I plan also to make cat_test and dog_test more similar.
(not sure I can work on that before next week)

Reply:
Sorry, I receive a *lot* of email so sometimes it's difficult to keep up with and respond to all GitHub comments (which are a lot of that email themselves)...

Using real images for the fake request would be somewhat reasonable, but only if they were checked in. Making real HTTP requests in tests is not ideal.